addons/im_livechat/static/src/embed/common/livechat_button.js

```diff
@@ -52,16 +52,36 @@ export class LivechatButton extends Component {
             animateNotification: !(
                 this.livechatService.thread || this.livechatService.shouldRestoreSession
             ),
+            hasAlreadyMovedOnce: false,
         });
         useMovable({
             cursor: "grabbing",
             ref: this.ref,
             elements: ".o-livechat-LivechatButton",
             onDrop: ({ top, left }) => {
+                if (!this.state.hasAlreadyMovedOnce) {
+                    this.state.hasAlreadyMovedOnce = true;
+                }
                 this.position.left = `${left}px`;
                 this.position.top = `${top}px`;
             },
         });
+        useExternalListener(document.body, "scroll", this._onScroll, { capture: true });
+    }
+
+    _onScroll(ev) {
+        if (!this.ref.el || this.state.hasAlreadyMovedOnce) {
+            return;
+        }
+        const wrappedElement = ev.target;
+        if (
+            wrappedElement.scrollHeight - wrappedElement.scrollTop ===
+            wrappedElement.clientHeight
+        ) {
+            this.ref.el.style.top = `calc(93% - ${LIVECHAT_BUTTON_SIZE}px)`;
+        } else {
+            this.ref.el.style.top = `calc(97% - ${LIVECHAT_BUTTON_SIZE}px)`;
+        }
```

Comment:
- We should avoid writing directly to the element; use state.position instead to prevent multiple sources of truth about the button position.
- I'd also rather have a ternary here instead of this branching; it makes things easier to read, IMO.
- Finally, I'd rename `wrappedElement` to `container`. The concept is easier to grasp and the name is shorter.

Reply:
I've commited a new proposal if you have time to review it!